met-api/tests/unit/api/test_widget.py

```diff
@@ -34,6 +36,81 @@ def test_create_widget(client, jwt, session, widget_info):  # pylint:disable=unu
                      headers=headers, content_type='application/json')
     assert rv.status_code == 200
 
+    rv = client.get('/api/widgets/engagement/' + str(engagement.id),
+                    headers=headers, content_type='application/json')
+    assert rv.status_code == 200
+    assert rv.json.get('result')[0].get('sort_index') == 1
+
+
+def test_create_widget_sort(client, jwt, session):  # pylint:disable=unused-argument
```

Comment:
Too many scenarios in one test? Can they be separated? Failures here is harder to track to what scenario failed. One  scenario for each test preferred

Reply:
I can separate the invalid test to another one.. the other test just asserts the sorting order

Reply:
We are posting widget_info_1 and widge_info_2 here. We don't need to do that right?

If we are testing the sorting, then the sorting endpoint should be the only one called. The test has multiple API calls, it is a bit difficult to understand what it is testing. Ideally, One test is testing only one API call. Everything else should be given.

Reply:
Will work on this later..I will create a ticket for tests in model layer and separate the tests